### ISSUE TITLE
Need help debugging results on MALACH corpus.

### ISSUE THREAD

**Issue by picheny-nyu** [espnet/espnet#2731]
I modified the AMI recipe to work with the Malach speech recognition corpus. The Malach speech recognition corpus was released in 2019 with files in Kaldi format along with a Kaldi recipe.

With Kaldi's chain model, we get ~24% WER.

With my modification to the AMI model I get a 74% WER.

Sum/Avg     |    809         12256    |   42.2         54.3         3.5         16.2        74.0         80.6    |

There are many insertions/repetitions of words like "THE":

REF:  *** **** **** ***** *** *** IT   CANNOT HAPPEN AGAIN THE  SAME  WAY BUT   i TELL HIM    to WATCH VERY CAREFULLY where **** **** *** they GO   WHERE they **** ***** **** ARE   and *** *** *** *** MOSTLY REMEMBER the BROTHERS SO  ONE CAN ALWAYS HELP the OTHER THIS WHAT THEY DID AND BECAUSE I   TOLD THEM I   YOUR MOTHER and *** *** MYSELF WE  WON'T BE  THERE FOREVER AND THAT'S ALL  there IS   BUT FOR the FOR   the WORLD THERE'S NOTHING TO  SAY IT'S SO  CRAZY THAT ANYTHING CAN HAPPEN and ***** WE'RE AT    WAR  AGAIN 

HYP:  AND THEY SAID THERE ARE AND THEY SAID   WHEN   WE    WERE THERE AND WHERE i **** WANTED to ***** HAVE ANYTHING  where THEY CAME AND they WERE THERE they WERE THERE WERE THERE and THE THE THE THE THE    THE      the THE      THE THE THE THE    THE  the THE   THE  THE  THE  THE THE THE     THE THE  THE  THE THE  THE    and THE THE THE    THE THE   THE THE   FOR     THE WAR    THAT there WERE THE THE the THING the ***** THE     THE     THE THE THE  THE THE   THE  THE      AND THING  and THERE AND   THERE WERE AND  

I am sure I am doing something horribly wrong but since I am not very familiar with the ESPNET code base I need some suggestions as to where to look. Help would be greatly appreciated!

Thanks
Michael

**Comment by sw005320** [comment]
I'm expecting that training would not be normally done.
We may need to tune some optimizer related parameters (but usually we can just decrease https://github.com/espnet/espnet/blob/master/egs2/ami/asr1/conf/tuning/train_asr_transformer.yaml#L38).
Could you paste the learning curve obtained by tensorboard or `egs2/xxx/asr1/exp/asr_train_*/images/acc.png`?

**Comment by picheny-nyu** [comment]
![acc](https://user-images.githubusercontent.com/55798528/100811254-2e884f80-3408-11eb-8de0-53c2fbf1b0e8.png)

See above for learning curve (acc.png)  - please tell me what else I can show you. Everything I see looks fine....

**Comment by sw005320** [comment]
Year, it looks normal.
Could you paste the training and decoding configs?

**Comment by picheny-nyu** [comment]
[config.pdf](https://github.com/espnet/espnet/files/5626177/config.pdf)
[decode.pdf](https://github.com/espnet/espnet/files/5626178/decode.pdf)
[train_asr.pdf](https://github.com/espnet/espnet/files/5626179/train_asr.pdf)

**Comment by sw005320** [comment]
OK, I think I got it. 
The validation acc started to be degraded in the very early stage (epoch 7), which happens when the learning rate is too large in the transformer case.
Could you reduce the learning rate (say 5?) in https://github.com/espnet/espnet/blob/master/egs2/ami/asr1/conf/tuning/train_asr_transformer.yaml#L38?

You don't have to finish all training to check this.
You can just check whether we have such early-stage degradation (if you observe it again you may further decrease the lr).

**Comment by picheny-nyu** [comment]
After 6 epochs it does not look very different than the above picture. Did I change the wrong parameter somehow?

![acc5](https://user-images.githubusercontent.com/55798528/100877393-b5741100-3476-11eb-9f06-897a4088ef14.png)
[config5.pdf](https://github.com/espnet/espnet/files/5629499/config5.pdf)

**Comment by sw005320** [comment]
For the transformer, we need to train more than 20 epochs (it depends on the amount of training data).
So, can you keep training it for at least up to 20 epochs (probably more)?
I feel that now training looks normal.

BTW, we usually average models during inference, but I found that the ami recipe does not seem to do it.
Once you fix this issue, you can change this parameter https://github.com/espnet/espnet/blob/master/egs2/ami/asr1/conf/tuning/train_asr_transformer.yaml#L50 from 1 to 5 (averaging 5 models).

**Comment by picheny-nyu** [comment]
Yes I will keep training. Our cluster is going down this morning for a few hours, does this code automatically restart at the right iteration or do I need to specify the iteration to start? 

Also, the ami data seems to be much better behaved than the malach data (will post a pic when the cluster comes back up). The amount of data is similar. 

BTW thanks for all your help. I would not have a chance to debug this myself!

**Comment by sw005320** [comment]
> Yes I will keep training. Our cluster is going down this morning for a few hours, does this code automatically restart at the right iteration or do I need to specify the iteration to start?

If you just re-run it from stage 10 (`./run.sh --stage 10 xxxx`), it automatically resumes training.

> Also, the ami data seems to be much better behaved than the malach data (will post a pic when the cluster comes back up). The amount of data is similar.

How about training accuracy behavior?
If it is similar, we'd have some overtraining and we can avoid it by the optimizer tuning or other regularization techniques.
If the training accuracy is also different, we might have some issues in the data preparation or simply this task is more difficult than ami.

> BTW thanks for all your help. I would not have a chance to debug this myself!

Sure, we're happy to help you.

**Comment by picheny-nyu** [comment]
Learning rate of 5.0 did not do much, it stopped after 8th epoch rather than 7th and WER got very slightly better. Trying lr of 2.5 and set model averging to 5 as you described above.

**Comment by picheny-nyu** [comment]
Changing learning rate to 2.5 cause it to run about 13 epochs and averaging over 5 models further reduced WER to 60% but it is still very high (and still many insertions of short words). I now lowered the lr to 1.25. However I wondered if the following issues could be causing problems:

a) The lengths of items in the test data are different than those in training data. In particular there are a number of long sentences which seem to cause a lot of problems in the test data:

<img width="400" alt="Screen Shot 2020-12-04 at 1 15 56 PM" src="https://user-images.githubusercontent.com/55798528/101284051-16b81f00-37ac-11eb-9680-360a289eb8c6.png">
<img width="390" alt="Screen Shot 2020-12-04 at 1 16 12 PM" src="https://user-images.githubusercontent.com/55798528/101284054-16b81f00-37ac-11eb-8f5b-40646022bd2b.png">
